components/count-badge/count-badge-mixin.js

```diff
@@ -0,0 +1,196 @@
+			},
+			/**
+			 * Optionally add a tooltip on the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			hasTooltip: {
+				type: Boolean,
+				attribute: 'has-tooltip'
+			},
+			/**
+			 * Optionally choose to not render the count badge when the number is zero. Defaults to false.
+			 * @type {boolean}
+			 */
+			hideZero: {
+				type: Boolean,
+				attribute: 'hide-zero'
+			},
+			/**
+			 * Optionally specify a digit limit, after which numbers are truncated. Defaults to two for "notification" type and no limit for "count" type.
+			 * @type {number}
+			 */
+			maxDigits: {
+				type: Number,
+				attribute: 'max-digits'
+			},
+			/**
+			 * The number to be displayed on the badge. Must be a positive integer.
+			 * @type {number}
+			 */
+			number: {
+				type: Number,
+				attribute: 'number'
+			},
+			/**
+			 * The size of the badge. Defaults to "small".
+			 * @type {'small'|'large'}
+			 */
+			size: {
+				type: String,
+				reflect: true,
+				attribute: 'size'
+			},
+			/**
+			 * Optionally choose to add a tab stop to the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			tabStop: {
+				type: Boolean,
+				attribute: 'tab-stop'
+			},
+			/**
+			 * * Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled.
+			 * @type {string}
+			 */
+			text: {
+				type: String
+			},
+			/**
+			 * The type of the badge. Defaults to "count".
+			 * @type {'count'|'notification'}
+			 */
+			type: {
+				type: String,
+				reflect: true,
+				attribute: 'type'
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this.announceChanges = false;
+		this.hasTooltip = false;
+		this.hideZero = false;
+		this.size = 'small';
+		this.tabStop = false;
+		this.text = '';
+		this.type = 'count';
+
+		this._badgeId = getUniqueId();
+		this._textId = getUniqueId();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.maxDigits && this.type === 'notification') {
+			// default to two digits for notification type
+			this.maxDigits = 2;
+		}
+	}
+
+	renderCount(tooltipElement, forceTooltipOn) {
+		let numberString = `${this.number}`;
+		const hideNumber = this.hideZero && this.number === 0;
+		if (hideNumber) {
+			numberString = '';
+		}
+		const numberVisibility = {
+			visibility: hideNumber ? 'hidden' : 'visible'
+		};
+		if (this.maxDigits && this.number.toString().length > this.maxDigits) {
+			numberString = `${'9'.repeat(this.maxDigits)}`;
+			numberString = this.localize('components.count-badge.plus', { number: numberString });
+		}
+		return html`
+			<div 
+			class="d2l-count-badge-wrapper"
+			id="${ifDefined(tooltipElement ? undefined : this._badgeId)}"
+			tabindex="${ifDefined((this.tabStop || this.hasTooltip) && (!hideNumber || tooltipElement) ? '0' : undefined)}" 
+			aria-labelledby="${ifDefined(this.hasTooltip ? undefined : this._textId)}"
+			role="${ifDefined(this.hasTooltip && !tooltipElement ? 'img' : undefined)}">
```

Comment:
The axe tests were failing with the tooltip saying that you can't have aria-labelledby without a role... I've added img here since that seemed to match the most. I also tried status, but that adds a live region and other behaviour we don't want here. Img doesn't add anything to screenreaders other than "graphic" in front of the description.

Reply:
Cool, I was actually wondering about that and how this "thing" would appear semantically. Although not perfect, image seems reasonable.